### PR TITLE
HPCC-19604 Spark access to super files

### PR DESCRIPTION
**Pull request by johnholt**
Extended HpccPart to accept an array of DataPartitions, each DataPartition mapped to a singler THOR file pert.

Extended HpccRemoteFileReader to work with an array of DataPartitions.

Extended test code for multiple data partitions.

Signed-off-by: johnholt <john.d.holt@lexisnexisrisk.com>

### CODE REVIEW

**Comment by richardkchapman** [comment]
@rpastrana Please review

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccPart.java [hunk 1]
```diff
@@ -47,17 +42,33 @@
    * @param part_ordinal the ordinal position of this part
    * @param part the data partition
    */
-  private HpccPart(int parts, int part_ordinal, DataPartition part) {
-    this.dataPart = part;
+  private HpccPart(int parts, int part_ordinal, DataPartition[] dataPartsIn) {
+    this.dataParts = new DataPartition[dataPartsIn.length];
+    for (int i=0; i<dataPartsIn.length; i++) this.dataParts[i] = dataPartsIn[i];
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccPart.java]: this is fine, but could we have have used this.dataParts = Arrays.copyOf(dataPartsIn,dataPartsIn.length) ?

File: DataAccess/src/main/java/org/hpccsystems/spark/HpccPart.java [hunk 2]
```diff
@@ -47,17 +42,33 @@
...
 
   /**
-   * Partition information
-   * @return information
+   * Partition information list.  A copy of the array.
+   * @return information list
    */
-  public DataPartition getPartitionInfo() { return this.dataPart; }
+  public DataPartition[] getPartitionInfoList() {
+    DataPartition[] rslt = new DataPartition[this.dataParts.length];
```
Comment (rpastrana) [inline DataAccess/src/main/java/org/hpccsystems/spark/HpccPart.java]: same comment as above regarding array copy

**Comment by johnholt** [comment]
@rpastrana  Changed explicit allocate&copy to Arrays.copyOf(...).  Added comment to better explain recurseFDI(...) which is a very descriptive name of what it is doing.

**Pull request merged**